pkg/nvlib/device/identifier.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package device
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// Identifier represents a GPU device in the form of an index or UUID
+type Identifier string
+
+// IsGpuIndex checks if an identifier is a full GPU index
+func (i Identifier) IsGpuIndex() bool {
+	if _, err := strconv.ParseUint(string(i), 10, 0); err != nil {
+		return false
+	}
+	return true
+}
+
+// IsMigIndex checks if an identifier is a MIG index
+func (i Identifier) IsMigIndex() bool {
+	split := strings.SplitN(string(i), ":", 2)
+	if len(split) != 2 {
+		return false
+	}
+	for _, s := range split {
+		if _, err := strconv.ParseUint(s, 10, 0); err != nil {
+			return false
+		}
+	}
```

Comment:
What do we think of:
```suggestion
	for _, s := range split {
		if !Identifier(s).IsGPUIndex() {
	        	return false
	        }
	}
```